feat: Add global repository search to GitHub node

ryanjhermes opened the pull request:
# Add Global Repository Search to GitHub Node

## Description
This PR adds a new "Search" resource to the GitHub node that allows users to search repositories globally across all of GitHub using the GitHub Search API.

## Changes Made

### Core Implementation
- **Added "Search" resource** to the GitHub node resource options
- **Added "Search Repositories" operation** with comprehensive parameters:
  - `query`: Search query string (supports GitHub search syntax)
  - `sort`: Sort options (best-match, stars, forks, help-wanted-issues, updated)
  - `order`: Sort order (asc/desc)
  - `returnAll`: Boolean to control pagination
  - `limit`: Maximum results when returnAll is false
- **Added execution logic** to call GitHub's `/search/repositories` API endpoint
- **Updated parameter visibility** to hide owner/repository fields for global search

### Testing
- **Unit Tests**: Created `GithubSearch.test.ts` with 6 comprehensive test cases
- **Workflow Tests**: Created `GithubSearchWorkflow.json` for workflow testing
- **Test Coverage**: Tests cover basic queries, returnAll functionality, different sort options, user filters, organization filters, and error handling

## Use Cases
This feature enables users to:
- Search for repositories by language, stars, forks, etc.
- Filter by user or organization: `user:facebook react`
- Find trending repositories: `language:javascript stars:>1000`
- Discover projects by topic: `org:microsoft language:typescript`
- Search for help-wanted repositories: `help-wanted-issues:>5`

## API Compatibility
- Uses official GitHub Search API (`/search/repositories`)
- Supports all GitHub search syntax and filters
- Handles pagination correctly with `returnAll` parameter
- Maintains existing authentication methods (Access Token/OAuth2)

## Testing
All tests pass:
```bash
- search repositories with basic query
-search repositories with returnAll enabled  
- handle search with different sort options
- handle search with user filter
- handle search with organization filter
- handle API errors gracefully
```

## Breaking Changes
None. This is a purely additive feature that doesn't affect existing functionality.

## Checklist
- [x] Tests included for new functionality
- [x] Code follows n8n conventions
- [x] Small, focused PR
- [x] Not a new node (enhancement to existing GitHub node)
- [x] Documentation updated (if applicable)

ryanjhermes committed:
feat: Add global repository search to GitHub node

Joffcom commented:
Hey @ryanjhermes,

Thank you for your contribution. We appreciate the time and effort you’ve taken to submit this pull request.

Before we can proceed, please ensure the following:
	•	Tests are included for any new functionality, logic changes or bug fixes.
	•	The PR aligns with our contribution guidelines.

Regarding new nodes:
We no longer accept new nodes directly into the core codebase. Instead, we encourage contributors to follow our Community Node Submission Guide to publish nodes independently.

If your node integrates with an AI service that you own or represent, please email nodes@n8n.io and we will be happy to discuss the best approach.

About review timelines:
This PR has been added to our internal tracker as "GHC-3511". While we plan to review it, we are currently unable to provide an exact timeframe. Our goal is to begin reviews within a month, but this may change depending on team priorities. We will reach out when the review begins.

Thank you again for contributing to n8n.

ryanjhermes committed:
Update packages/nodes-base/nodes/Github/__tests__/node/GithubSearchWorkflow.json

Co-authored-by: cubic-dev-ai[bot] <191113872+cubic-dev-ai[bot]@users.noreply.github.com>

ryanjhermes committed:
Update packages/nodes-base/nodes/Github/Github.node.ts

Co-authored-by: cubic-dev-ai[bot] <191113872+cubic-dev-ai[bot]@users.noreply.github.com>

ryanjhermes committed:
fix: Fix Github node by updating search parameters to comply with GitHub API limits

ryanjhermes committed:
refactor: Simplify type assertion for query parameter

ryanjhermes commented:
Hey @Joffcom! 

Thank you for the feedback, I've addressed the issues identified in the review and would like to confirm that this PR meets all requirements. All tests pass, and the feature is ready for review and addresses all the technical concerns raised in the previous feedback. Thank you for your time!
